src/Settings.cpp

```diff
@@ -0,0 +1,339 @@
+    m_enableWOL = false;
+
+  if (!XBMC->GetSetting("woltimeout", &m_timeoutWOL))
+    m_timeoutWOL = 20;
+
+  if (!XBMC->GetSetting("guideartwork", &m_downloadGuideArtwork))
+    m_downloadGuideArtwork = DEFAULT_GUIDE_ARTWORK;
+
+  if (!XBMC->GetSetting("remoteaccess", &m_remoteAccess))
+    m_remoteAccess = false;
+
+  if (!XBMC->GetSetting("flattenrecording", &m_flattenRecording))
+    m_flattenRecording = false;
+
+  if (!XBMC->GetSetting("reseticons", &m_resetIcons))
+    m_resetIcons = false;
+
+  if (!XBMC->GetSetting("kodilook", &m_kodiLook))
+    m_kodiLook = false;
+
+  if (!XBMC->GetSetting("prebuffer", &m_prebuffer))
+    m_prebuffer = 8;
+
+  if (!XBMC->GetSetting("prebuffer5", &m_prebuffer5))
+    m_prebuffer5 = 0;
+
+  if (!XBMC->GetSetting("chunklivetv", &m_liveChunkSize))
+    m_liveChunkSize = 64;
+
+  if (!XBMC->GetSetting("chunkrecording", &m_chunkRecording))
+    m_chunkRecording = 32;
+
+  if (XBMC->GetSetting("resolution", &buffer))
+    m_resolution = buffer;
+  else
+    m_resolution = "720";
+
+/* Log the current settings for debugging purposes */
+  XBMC->Log(LOG_DEBUG, "settings: host='%s', port=%i, mac=%4.4s...", m_hostname.c_str(), m_port, m_hostMACAddress.c_str());
+
+}
+
+ADDON_STATUS Settings::ReadBackendSettings()
+{
+  // check server version
+  std::string settings;
+  if (NextPVR::m_backEnd->DoRequest("/service?method=setting.list", settings) == HTTP_OK)
+  {
+    TiXmlDocument settingsDoc;
+    if (settingsDoc.Parse(settings.c_str()) != NULL)
+    {
+      //XBMC->Log(LOG_NOTICE, "Settings:\n");
+      //dump_to_log(&settingsDoc, 0);
+      if (XMLUtils::GetInt(settingsDoc.RootElement(), "NextPVRVersion", m_backendVersion))
+      {
+        // NextPVR server
+        XBMC->Log(LOG_INFO, "NextPVR version: %d", m_backendVersion);
+
+        // is the server new enough
+        if (m_backendVersion < 40204)
+        {
+          XBMC->Log(LOG_ERROR, "Your NextPVR version '%d' is too old. Please upgrade to '%s' or higher!", m_backendVersion, NEXTPVRC_MIN_VERSION_STRING);
+          XBMC->QueueNotification(QUEUE_ERROR, XBMC->GetLocalizedString(30050));
+          XBMC->QueueNotification(QUEUE_ERROR, XBMC->GetLocalizedString(30051), NEXTPVRC_MIN_VERSION_STRING);
+          return ADDON_STATUS_PERMANENT_FAILURE;
+        }
+      }
+
+      // load padding defaults
+      m_defaultPrePadding = 1;
+      XMLUtils::GetInt(settingsDoc.RootElement(), "PrePadding", m_defaultPrePadding);
+
+      m_defaultPostPadding = 2;
+      XMLUtils::GetInt(settingsDoc.RootElement(), "PostPadding", m_defaultPostPadding);
+
+      m_showNew = false;
+      XMLUtils::GetBoolean(settingsDoc.RootElement(),"ShowNewInGuide",m_showNew);
+
+      std::string recordingDirectories;
+      if (XMLUtils::GetString(settingsDoc.RootElement(),"RecordingDirectories",recordingDirectories))
+      {
+        m_recordingDirectories = StringUtils::Split(recordingDirectories,",",0);
+        /*
```

Comment:
Remove commented code or add a TODO:

Reply:
Actually that was the old code left in to remind myself to test that Split did work as expected and I forgot to remove it.  Done